modin/modin_logging/config.py

```diff
@@ -0,0 +1,173 @@
+        records timestamps at the microsecond level of granularity.
+
+        Parameters
+        ----------
+        record: LogRecord
+            The specified LogRecord object.
+        datefmt: str, default: None
+            Used with time.ststrftime() to format time record.
+
+        Returns
+        -------
+        datetime
+            datetime object containing microsecond timestamp.
+        """
+
+        ct = dt.datetime.fromtimestamp(record.created)
+        if datefmt:
+            s = ct.strftime(datefmt)
+        else:
+            # Format datetime object ct to microseconds
+            t = ct.strftime("%Y-%m-%d %H:%M:%S")
+            s = f"{t},{record.msecs:03}"
+        return s
+
+
+def bytes_int_to_str(num_bytes, suffix="B"):
+    """
+    Scale bytes to its human-readable format (e.g: 1253656678 => '1.17GB').
+
+    Parameters
+    ----------
+    num_bytes: int
+        Number of bytes.
+    suffix: str, default: "B"
+        Suffix to add to conversion of num_bytes.
+
+    Returns
+    -------
+    str
+        Human-readable string format.
+    """
+
+    factor = 1000
+    if num_bytes > 1000000000000000000:
+        raise ValueError("System memory exceeds expectations")
+    for unit in ["", "K", "M", "G", "T", "P"]:
+        if num_bytes < factor:
+            return f"{num_bytes:.2f}{unit}{suffix}"
+        num_bytes /= factor
+
+
+def configure_logging(level):
+    """
+    Configure Modin logging by setting up directory structure and formatting.
+
+    Parameters
+    ----------
+    level: str
+        The log level (logging.INFO, logging.DEBUG).
+    """
+
+    global __LOGGER_CONFIGURED__
+    logger = logging.getLogger("modin.logger")
+    job_id = uuid.uuid4().hex
+    log_filename = f".modin/logs/job_{job_id}.log"
+
+    os.makedirs(os.path.dirname(log_filename), exist_ok=True)
+
+    logfile = logging.FileHandler(log_filename, "a")
+    formatter = ModinFormatter(
+        fmt="%(process)d, %(thread)d, %(asctime)s, %(message)s",
+        datefmt="%Y-%m-%d,%H:%M:%S.%f",
+    )
+    logfile.setFormatter(formatter)
+    logger.addHandler(logfile)
+    logger.setLevel(level)
+
+    __LOGGER_CONFIGURED__ = True
+
+
+def memory_thread(logger, sleep_time):
+    """
+    Configure Modin logging system memory profiling thread.
+
+    Parameters
+    ----------
+    logger: logging.Logger
+        The logger object.
+    sleep_time: int
+        The interval at which to profile system memory.
+    """
+
+    while True:
+        svmem = psutil.virtual_memory()
+        logger.info(f"Memory Percentage: {svmem.percent}%")
+        time.sleep(sleep_time)
+
+
+def get_logger():
+    """
+    Configures Modin logger based on Modin config and returns the logger.
+
+    Returns
+    -------
+    logging.Logger
+        The Modin logger.
+    """
+
+    if not __LOGGER_CONFIGURED__ and LogMode.get() != "disable":
+        if LogMode.get() == "enable_api_only":
```

Comment:
I prefer to factor this out into a separate function and call it separately since it should only be executed at the start.